src/components/Collapse.tsx

```diff
@@ -1,58 +1,60 @@
-import React, { PropsWithChildren } from 'react';
+import React, { PropsWithChildren, ReactNode } from 'react';
 import { GrafanaTheme2 } from '@grafana/data';
 import { Icon, useStyles2 } from '@grafana/ui';
 import { css, cx } from '@emotion/css';
 
 interface Props {
+  className?: string;
   isOpen?: boolean;
-  label: string | JSX.Element;
-  loading?: boolean;
-  collapsible?: boolean;
+  label: ReactNode;
   onToggle?: (isOpen: boolean) => void;
 }
 
-const getStyles = (theme: GrafanaTheme2) => ({
-  container: css`
-    border-left: none;
-    border-right: none;
-    border-bottom: none;
-    margin-bottom: 0;
-    padding: ${theme.spacing(2)} 0;
-  `,
-  header: css`
-    display: flex;
-    align-items: center;
-    transition: all 0.1s linear;
-    cursor: pointer;
-  `,
-  headerExpanded: css`
-    padding-bottom: ${theme.spacing(1)};
-  `,
-  headerIcon: css`
-    margin-right: ${theme.spacing(1)};
-  `,
-  label: css`
-    margin-right: ${theme.spacing(1)};
-    font-size: ${theme.typography.h4.fontSize};
-  `,
-  body: css`
-    padding-top: ${theme.spacing(1)};
-  `,
-  hidden: css`
-    display: none;
-  `,
-});
-
-export const Collapse = ({ isOpen, label, children, onToggle, ...props }: PropsWithChildren<Props>) => {
+export const Collapse = ({ className, isOpen, label, children, onToggle, ...props }: PropsWithChildren<Props>) => {
   const styles = useStyles2(getStyles);
 
   return (
-    <div className={cx(['panel-container', styles.container])}>
-      <div className={styles.header} onClick={() => onToggle && onToggle(Boolean(isOpen))}>
+    <div className={cx(styles.container, className)} {...props}>
+      <button
+        aria-expanded={isOpen}
+        className={styles.header}
+        onClick={() => onToggle && onToggle(Boolean(isOpen))}
+        type="button"
+      >
         <div className={styles.label}>{label}</div>
         <Icon name={isOpen ? 'angle-down' : 'angle-right'} className={styles.headerIcon} />
-      </div>
+      </button>
       <div className={cx(styles.body, { [styles.hidden]: !isOpen })}>{children}</div>
     </div>
   );
 };
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  container: css({
```

Comment:
updated `css` syntax. nice!

Reply:
Yeah! I found a super weird bug early on where one property specifically worked with the object syntax rather than the template literal one, then I remembered @ashharrison90 [had done a lot of work to convert ](https://github.com/grafana/grafana/pull/71374)`grafana/grafana` to the object syntax and his reasons why so I figured we should follow their example and have at it 😄